pkg/ocm/managedcluster.go

```diff
@@ -0,0 +1,201 @@
+package ocm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterV1Client "open-cluster-management.io/api/client/cluster/clientset/versioned/typed/cluster/v1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+)
+
+// ManagedClusterBuilder provides a struct for the ManagedCluster object containing connection to the cluster and the
+// ManagedCluster definitions.
+type ManagedClusterBuilder struct {
+	Definition *clusterv1.ManagedCluster
+	Object     *clusterv1.ManagedCluster
+	errorMsg   string
+	apiClient  clusterV1Client.ClusterV1Interface
+}
+
+// ManagedClusterAdditionalOptions additional options for ManagedCluster object.
+type ManagedClusterAdditionalOptions func(builder *ManagedClusterBuilder) (*ManagedClusterBuilder, error)
+
+// NewManagedClusterBuilder creates a new instance of ManagedClusterBuilder.
+func NewManagedClusterBuilder(apiClient *clients.Settings, name string) *ManagedClusterBuilder {
+	glog.V(100).Infof(
+		"Initializing new ManagedCluster structure with the following params: name: %s", name)
+
+	builder := &ManagedClusterBuilder{
```

Comment:
Please check if apiClient is nil and return nil if so

Reply:
I don't think any of the other packages return early on a nil apiClient in NewXXBuilder(), not sure if that should be made a global change similar to #355. I've changed it here though

Reply:
It's a requirement for any resources that have had their apiClient limited ([example](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/lca/seedgenerator.go#L43-L47)). Otherwise, you can cause a panic on line 40

Reply:
Ah makes sense, I'll go ahead and update the cgu package then since I think it'd apply there too